packages/react-ui/src/hooks/useImperativeIntl.js

```diff
@@ -0,0 +1,34 @@
+import { createIntl, createIntlCache } from 'react-intl';
+import { messages } from '../localization';
+import { useMemo } from 'react';
+import { match } from '@formatjs/intl-localematcher';
+import { flattenMessages } from '../localization/localeUtils';
+
+export const DEFAULT_LOCALE = 'en';
+const cache = createIntlCache();
+
+export default function useImperativeIntl(intlConfig) {
+  const locale = intlConfig?.locale || DEFAULT_LOCALE;
+  const messagesLocale = findMatchingMessagesLocale(locale);
+  const intMessages = intlConfig?.messages
+    ? flattenMessages(intlConfig?.messages)
+    : messages[messagesLocale];
```

Comment:
i think all lines above should be in memo together with `createIntl`

mainly `findMatchingMessagesLocale` and `flattenMessages` are not trivial and everything depends _only_ on `intlConfig`

Reply:
Done